internal/services/disbursement_management_service.go

```diff
@@ -210,19 +220,88 @@ func (s *DisbursementManagementService) StartDisbursement(ctx context.Context, d
 			}
 		}
 
-		// 4. Update all correct payment status to `ready`
+		// 4. Check if there is enough balance from the distribution wallet for this disbursement along with any pending disbursements
+		rootAccount, err := s.horizonClient.AccountDetail(
+			horizonclient.AccountRequest{AccountID: distributionPubKey})
+		if err != nil {
+			return fmt.Errorf("error getting details for root account from horizon client: %w", err)
+		}
+
+		var distributionBalance float64
+		for _, b := range rootAccount.Balances {
+			if b.Asset.Code == disbursement.Asset.Code && b.Asset.Issuer == disbursement.Asset.Issuer {
+				distributionBalance, err = strconv.ParseFloat(b.Balance, 64)
+				if err != nil {
+					return fmt.Errorf("error converting Horizon distribution account balance %s into float: %w", b.Balance, err)
+				}
+			}
+		}
+
+		disbursementAmount, err := strconv.ParseFloat(disbursement.TotalAmount, 64)
+		if err != nil {
+			return fmt.Errorf(
+				"error converting total amount %s for disbursement id %s into float: %w",
+				disbursement.TotalAmount,
+				disbursementID,
+				err,
+			)
+		}
+
+		var totalPendingAmount float64 = 0.0
+		incompletePayments, err := s.models.Payment.GetAll(ctx, &data.QueryParams{
+			Filters: map[data.FilterKey]interface{}{
+				data.FilterKeyStatus: []data.PaymentStatus{
+					data.ReadyPaymentStatus,
+					data.PendingPaymentStatus,
+					data.PausedPaymentStatus,
+				},
+			},
+		}, dbTx)
+		if err != nil {
+			return fmt.Errorf("error retreiving incomplete payments: %w", err)
+		}
+
+		for _, ip := range incompletePayments {
+			if ip.Disbursement.ID == disbursementID {
+				continue
+			}
+
+			paymentAmount, parsePaymentAmountErr := strconv.ParseFloat(ip.Amount, 64)
+			if parsePaymentAmountErr != nil {
+				return fmt.Errorf(
+					"error converting amount %s for paymment id %s into float: %w",
+					ip.Amount,
+					ip.ID,
+					err,
+				)
+			}
+			totalPendingAmount += paymentAmount
+		}
+
+		if (distributionBalance - (disbursementAmount + totalPendingAmount)) < 0 {
+			log.Ctx(ctx).Errorf(
+				"Insufficient distribution account balance %f to fulfill amount %f for disbursement id %s and total pending amount %f",
+				distributionBalance,
+				disbursementAmount,
+				disbursementID,
+				totalPendingAmount,
+			)
+			return fmt.Errorf("error starting disbursement with id %s: insufficient balance on distribution account", disbursementID)
+		}
```

Comment:
👀   When you return the error in this manner, how is it being communicated to the frontend caller?

Reply:
so when the validation fails, the handler returns the error to the web client in the response but only the message is displayed. would this need to a change on the front end?

```
default:
  msg := fmt.Sprintf("Cannot update disbursement ID %s with status: %s", disbursementID, toStatus)
  httperror.InternalError(ctx, msg, err, nil).Render(w)
```

<img width="652" alt="Screenshot 2024-01-30 at 4 31 17 PM" src="https://github.com/stellar/stellar-disbursement-platform-backend/assets/20179122/53ae9c44-c29c-4993-801c-2a6eb35b614d">

Reply:
500 is usually for unexpected internal errors that are out of our control though. I don't think a 500 is the best option here. 

Maybe 400, 409 or another one would fit better, WDYT?

Reply:
The other validations seem to be generating 400s for the most part

Reply:
Ok, I changed it to a 409 since I think that makes the most sense!

Reply:
That makes sense to me, TY!